### ISSUE TITLE
Atom detected as malware by antiviruses

### ISSUE THREAD

**Issue by achalatsis** [atom/atom#3927]
Atom version: 0.136.0

From atom 0.136.0, the following two files are falsely detected as: Win32:Malware-gen:
atom-0.136.0\build\windows\Setup.exe 
atom-0.136.0\build\windows\Update.exe 

Avast version: 2015.10.0.2206
Virus signatures version: 141021-0

I will also submit this to avast.

**Comment by jeffjarchow** [comment]
squirrel.exe was just removed from my system today by Symantec.cloud. It is their endpoint protection software. Luckily with Symantec it is an easy restore. I have been using atom for a few weeks now and this is the first time it has been reported as a suspicious item.

Version info:
```
Symantec.cloud - Cloud Agent	2.03.61.2573
Symantec.cloud - Endpoint Protection	NIS-21.5.0.19
```

Symantec Report:
```
Threat Name: squirrel.exe
Threat Type: Suspicious item
File Name: c:\users\username\appdata\local\atom\app-0.181.0\squirrel.exe
```

**Comment by BlaM** [comment]
Problem still exists.  (Same Symantec version as @jeffjarchow)

    Date & Time: 4/22/2015 3:00:39 PM
    Risk: High
    Threat: squirrel.exe (SONAR.Heuristic.120) detected by SONAR
    Filename: c:\users\ddeobald\appdata\local\atom\app-0.194.0\squirrel.exe

**Comment by Zireael07** [comment]
I had the similar "suspicious" reports from my Norton 360.
I've learned to ignore them and whitelist apps. It seems Norton likes to detect apps which connect online as "suspicious"...

**Comment by timdp** [comment]
I encountered this a few days ago when Atom updated to (I think) version 0.198.0. A while later, I read about Avast's latest update [flagging](https://forum.avast.com/index.php?topic=170705.0) Chrome as a virus. The latter never actually happened to me but there might be a link.

**Comment by Zireael07** [comment]
Addendum: my Norton 360 reported the same threat as @BlaM's Symantec. That was on 0.194 or 0.196. Didn't have any issues with the 0.198 update, though.

**Comment by paulcbetts** [comment]
Still passes Virustotal: https://www.virustotal.com/en/file/d10ee38fd3e4e71901e1a556f947531770adea07d691818921f11d99b9c64462/analysis/

**Comment by ryush00** [comment]
![photo_2015-06-06_23-14-37](https://cloud.githubusercontent.com/assets/4997174/8019930/cb244fa4-0ca1-11e5-978c-2b7e90e5212e.jpg)

**Comment by di48l069** [comment]
Hey guys how can I fix this see image...

![virus](https://cloud.githubusercontent.com/assets/7794777/8021037/15282960-0c91-11e5-9f6a-1f039b4546c2.PNG)

**Comment by ryush00** [comment]
Right click and click "restore and add to ignore list"
2015. 6. 7. 오전 4:16에 "di48l069" <notifications@github.com>님이 작성:

**Comment by Deraj82901** [comment]
I am having an issue like this on Linux, so I dont know if I should post it here, but im scanning it with Comodo, and it says core.log.js is infected with Heur.Dual.Extensions

**Comment by paulcbetts** [comment]
@di48l069 Uninstall Avast

**Comment by BlaM** [comment]
@Deraj82901 I can see why Comodo returns that message (it's .log.js, which may appear as .log in applications and is still kind of executable) - but that's definitely one of the most useless classifications I've seen. However: That's not the same "virus" detection as the main one here.

@di48l069 Avast seems to think that Atom.exe is bad - but the issue here is mainly about installation so maybe even your issue is not the same "virus" detection.

**Comment by danieldogeanu** [comment]
Guys, it's not Atom that's infected, it's your PC! Avast reported to me that atom.exe is infected with FileRepMetagen as I was working. Then it started reporting critical system files the same, and my MVIDIA driver was crashing constantly.

**Comment by BlaM** [comment]
@danieldogeanu 

1. As mentioned in the issues above...
  > files are falsely detected

2. *FileRepMetagen* is not an "exact" virus definition. Your Computer cannot be infected by "FileRepMetagen" (or *Heur.Dual.Extensions*, *SONAR.Heuristic.120* or similar), because FileRep is just the name of an algorithm used in Avast - not a piece of software/virus/trojan/whatever. 

  You can find out more about FileRepMetagen in the Avast Form: http://forum.avadas.de/threads/3920-avast!-FileRep (German).

  > List of attributes used to evaluate files "in the cloud" by FileRep:
  > * Age of File (Emergence)
  > * Number of computers that have executed the file (Prevalence)
  > * Distribution of the file
  > * Source URL
  > * Status of digital signatures

  So any file can be evaluated as "FileRepMetagen". There does not need to be an actual infection of any kind. It's enough if there are only few people using the same software.

And obviously antivirus software is doing a terrible job communicating what they are doing...

**Comment by Deraj82901** [comment]
Alrighty. Sorry for the off topic stuff! But this means that Atom is completely safe? It may help you to know that im running Ubuntu 14.04. I built Atom from source.

**Comment by danieldogeanu** [comment]
@BlaM Oh, I see. So my Avast freaking out was just an Avast problem. Thanks for pointing out. I should get a more serious antivirus.

**Comment by BlaM** [comment]
Avast is a serious antivirus - just TOO serious sometimes ;)

**Comment by distriker** [comment]
Anyone have received a reply from Avast to the false positive? It is very annoyance. When Avast show me the virus alert was totally surprised.

**Comment by ismay** [comment]
Don't know how much of a priority this has, but it's preventing my colleagues from installing atom (as they're on windows machines with virus protection and without administrator rights).

**Comment by BlaM** [comment]
Today this broke Atom by killing parts of the update package. Atom started upgrading and then the editor wouldn't start up after that. Best guess is that I had a "broken half update". Had to fix by redownloading the installer from atom.io

**Comment by teki** [comment]
Windows Defender reports it too now:

The following error occurred: Error code 0x80508023. The program could not find the malware and other potentially unwanted software on this computer. 

Category: Trojan Downloader

Description: This program is dangerous and downloads other programs.

Recommended action: Remove this software immediately.

Items: 
file:C:\Users\qqq\AppData\Local\atom\app-1.0.4\atom.exe
process:pid:6312,ProcessStart:130832981135644559

**Comment by paulcbetts** [comment]
Oof, that's really bad if Defender is detecting it. Are we not signing our executables anymore?

**Comment by filipesilva** [comment]
Getting it as well, win10 and atom 1.0.4. Wasn't happening yesterday.

**Comment by almostearthling** [comment]
> Wasn't happening yesterday

Maybe because of the Windows Defender update.

F.

**Comment by filipesilva** [comment]
Yeap, I think so.

**Comment by jarig** [comment]
And in corp. networks cause of that Sec. asks to wipe all laptops with it :\

**Comment by ismay** [comment]
@jarig Exactly, this is a major issue for corporate use.

**Comment by batjko** [comment]
Dudes... check if 1.0.5 fixes the issue?
https://github.com/atom/atom/releases/tag/v1.0.5

**Comment by jurosh** [comment]
@batjko Yes, atom is working again after running setup (Windows Defender)  :wink:

**Comment by thomasjo** [comment]
Judging from reports I've seen here, and on Slack, I believe this issue was resolved as of Atom v1.0.5.
If anyone has experience contrary to that assumption, please post a comment with details and we'll re-open the issue.

**Comment by mizchi** [comment]
I encountered this after published this app. [Kobito for Windows](http://kobito.qiita.com/win) 

[SONAR.SuspBeh!gen3 Removal - Removing Help | Symantec](http://www.symantec.com/security_response/writeup.jsp?docid=2013-111415-1843-99&tabid=3 "SONAR.SuspBeh!gen3 Removal - Removing Help | Symantec")

I reported symantec about this and they stopped alerting.

(I told this to @benogle directly at YAPC ASIA  yesterday)

**Comment by batjko** [comment]
I assume Symantec and other antivirs have formal procedures for developers/vendors to request exceptions for their apps?

**Comment by AaronTorgerson** [comment]
FWIW, I just downloaded the installer from https://atom.io and tried to install it on Windows 10, and Kaspersky identified it as a trojan and deleted it.

**Comment by paulcbetts** [comment]
I realize that this is still an issue Generally™, but seriously - there is no reason to use 3rd party antivirus with Windows 10, it is already built-in. Windows Defender is a full antivirus application from Windows 8 onward.

**Comment by ihorvorotnov** [comment]
Tried downloading Atom installer right now. Avast blocks the download, reporting it's infected (which, I believe, is not true). Turned off Avast. Now Defender blocked it... Damn, why I even decided to give it a try? ST3 does the job and never caused any issues. And ST devs/collabs never say smth like "uninstall avast" :)

**Comment by paulcbetts** [comment]
>  Now Defender blocked it.

Now that's interesting to me. Can you share a screenshot?

**Comment by ihorvorotnov** [comment]
@paulcbetts gave it another try today. Avast still blocks (see screenshot below), but it passed through Defender. IDK what was the problem with it yesterday. Maybe some Win glitch.

Installed, turned Avast back on - works fine so far.

![2015-10-14_15-42-35](https://cloud.githubusercontent.com/assets/1066083/10484015/53468a24-728b-11e5-8683-bc5dc0185437.jpg)

**Comment by paulcbetts** [comment]
> I suppose you're on *nix / Mac, not using Win on a daily basis and just heard somewhere on the internet that MS made a Defender a full-featured built-in antivirus software.

I worked on the Windows Kernel at Microsoft, code I wrote is literally built into your computer, tell me more about how I'm not a real Windows user

**Comment by benogle** [comment]
Now now. If either of these things think it's a virus, that is cause for concern. I thought we were passing through Avast at some point?

**Comment by paulcbetts** [comment]
@benogle It looks like Evo-Gen is the same kind of heuristic as the others - from http://malwaretips.com/blogs/win32evo-gen-susp-virus:

![](http://cl.ly/image/1E3M3n1Q2I14/Remove_Win32Evo-gen_Susp_by_Avast_Removal_Guide_2015-10-14_16-39-30.png)

**Comment by krimdomu** [comment]
Same for me with Kaspersky. It prevents the execution of AtomeSetup.exe because it thinks it is "PDM:Trojan.Win32.Generic"

**Comment by Traverse** [comment]
Since I had the same thing happen to me with Kaspersky I took the liberty to email them about it being a false positive, just received an email back from them and it should be fixed in the next update.

**Comment by paulcbetts** [comment]
@Traverse Thanks a lot! If other folx can do the same thing, that'd be amazing. Many AV vendors don't let you submit false positives without paying for their product :-/

**Comment by AlexLeung** [comment]
This just happened today with me and Norton 360

**Comment by jeffwareham** [comment]
I had this happen for the first time with the new update of Atom in Nov 2016.  As above, Norton flagged SONAR.AM.C!g1 as a Trojan/Virus.  I kept reading online that I should consider it a false positive, but when I finally removed it, the odd behavior (lagging processes, video-refresh fails, and the slow network behavior) all disappeared.  Not sure if it is a virus, but it certainly wrecked havoc with my system.

**Comment by TrendaFet** [comment]
@jeffwareham Same thing just happened with me. This issue has been opened for a very long time, I hope this is sorted out soon. My tinfoil hat paranoid side is stopping me from using Atom.

**Comment by 50Wliu** [comment]
In general this comes and goes as AV vendors tweak their virus-detection mechanisms.  For anyone who's experiencing this, please file a false positive report with your vendor.

**Comment by Polve** [comment]
Avira antivirus for windows detects it as malware.

**Comment by rezaamya** [comment]
okay!
then what we should do!?
it will detect as a virus and Avast will delete the exe file each time!

**Comment by Matt-F-95** [comment]
Avast just deleted Atom from my laptop, it said the infection was IDP.ARES.Generic and it was in the atom update.exe

Is this a false report?

**Comment by Swoy** [comment]
I can also report that Avast is complaining about a IDP.ARES.Generic infection. However, when running the file against VirusTotal, I get the following results:
![2017-08-17 07_46_22-antivirus scan for 0773548516b3ed1568f0dcc936810599574566e5c21c8ff53a8474a90768a](https://user-images.githubusercontent.com/5272077/29397666-385bb6ba-8320-11e7-806f-1e91e61747f6.png)

Where Avast has marked it as safe. Seems the behavior of that file is tagged as unsafe when monitored.

**Comment by sunsp1der** [comment]
Getting this same problem, though it just says IDP.Generic for me.

**Comment by Bru141** [comment]
Also got the continual attack warnings. I'm running "WinPatrolWAR" so I am able to see the executables as they launch and allow or Quarentine them. Foe example.. "$r61ixy4.exe (32bit)" or $rdqhyb4.exe(64bit) which were both launched from the atom setup.exe file located in the recycle bin. I also noticed my python program "Anaconda" was trying to run scripts.."anaconda3\scripts (2to3.exe)". When I looked in my Win 7 control panel to uninstall programs I noticed that there was a second program in there installed on the same date as the Atom IDE. It was a "win64 driver updater" made in China. I removed Atom and the Chinese program...And am now scanning for any potential problems with ESET. Thanks for any advice.
